tla-pp/src/test/scala/at/forsyte/apalache/tla/pp/TestNormalizer.scala

```diff
@@ -38,6 +38,30 @@ class TestNormalizer extends FunSuite with BeforeAndAfterEach {
     assert(expected == output)
   }
 
+  test("""~ IF x THEN y ELSE z ~~> IF x THEN ~y ELSE ~z""") {
+    val input = tla
+      .not(
+          tla.ite(
+              tla.name("x") ? "b",
+              tla.name("y") ? "b",
+              tla.name("z") ? "b"
```

Comment:
Let's produce types with the newer API, that is, `tla.name("x") as BoolT1()`. The benefit is that we do not have to carry a map of types around. We have not switched to the new API in all tests yet, so `? "foo"` appears in a lot of places.